lib/Backend/IR.cpp

```diff
@@ -4263,6 +4263,19 @@ GeneratorBailInInstr::New(IR::Instr* yieldInstr, Func* func)
     return labelInstr;
 }
 
+GeneratorBailInInstr* GeneratorBailInInstr::FromBailOutInfo(BailOutInfo* bailOutInfo)
+{
+    Assert(bailOutInfo);
+    IR::Instr* instr = bailOutInfo->bailOutInstr;
+    if (!instr || instr->m_opcode != Js::OpCode::Yield)
+    {
+        return nullptr;
+    }
+
+    Assert(instr->m_next && instr->m_next->m_next);
+    return instr->m_next->m_next->AsGeneratorBailInInstr();
```

Comment:
I feel like this is asking for trouble, maybe we want to guard this with `AssertOrFailFast(IsGeneratorBailInInstr())`, or even just do an early return if it isn't

Reply:
Yeah, I agree the `m_next->m_next` thing is icky. What would be a better way?

Clearly every `OpCode::Yield` must have a following `GeneratorBailIn` right? So it's just a matter of where in the forward list it is. If there's **isn't** one in the forward list (before the next `Yield`) then we have to fail fast.

So we could do this:

```cpp
for (auto* next = instr->m_next; next; next = next->m_instr)
{
    if (next->m_opcode === Js::OpCode::Yield) 
        break;
    else if (next->IsGeneratorBailInInstr())
        return next->AsGeneratorBailInInstr();
}
AssertOrFailFastMsg(false, "Yield not followed by GeneratorBailInInstr");
```

Do you think that would be better?